Finalized registration page details

MatthewGWilson opened the pull request:
Sorry if I screwed anything up with the pull request. Let me know if you need info or changes from me.

jameskokoska reviewed (changes requested):
- [x] If you removed reference to John Doe, should you remove the emergency contact placeholder jane doe?
- [x] Legal name should be split into first and last, as that is how we are asking for the name in the create account page, see https://github.com/UofT-Frosh-Orientation/orientation-website/pull/164
- [x] Legal name and preferred name still seem a bit redundant, it will most likely just replace the first and last name they have already entered when they made an account. Since Frosh can edit their data, see https://github.com/UofT-Frosh-Orientation/orientation-website/issues/180, it might not be needed to ask this again
- [x] Since `Prefer Not to Say` is capitalized, should `other` also be capitalized
- [ ] All radio buttons should have a default value (It is best practice, and there is currently no implementation of any check function on missing radio button fields not being selected)
- [x] When `Accomodation Contact Information` is set to email, you can't provide an email
- [x] `Accomodation` in Accomodation Contact Information is spelt wrong
- [x] `Carries 2 epi-pens` should Epi-Pen be capitalized as it is in the title?
- [x] Is `Your address won't be used except for delivery of the Kit.` accurate in the payment terms of service. Are Kits being delivered this year?
- [x] Should `Where will you be located for the majority of the summer?` be a dropdown, Where will you be located in the summer? with GTA (Toronto) and Other. When other selected make this `If you are not in the GTA already, approximately when are you planning to move to Toronto?` fillable. It will logically make more sense this way and be easier to sort by people located in the GTA or not (because people may have different ways of entering they are in the GTA, such as saying Mississauga, Etobicoke, Toronto, Markham etc
- [ ] `How do you plan on commuting to campus?` and `Other Commuter Option` should be next to each other (or `Other Commuter Option` and `Which stop do you get on for the subway or go train?` should be beside each other to minimize visual space)
- [ ] Do people know what Skule Patrol is? maybe find a more common name
- [ ] Most fields lack an easy to understand error message, most just say `Missing field`
- [ ] Right now, the area code is not required for phone numbers, is this okay?
- [ ] When `Accomodation Contact Information` Phone is selected, a phone input should be available. Information can be sorted easier when different input fields are used

Tickets have been created for the following issues:

- [ ] Placeholder text should be a lighter color, I keep thinking it's filled out with wrong information visually even though it is just a placeholder https://github.com/UofT-Frosh-Orientation/orientation-website/issues/189
- [x] Disabled accordion should have `cursor: not-allowed;` as it still seems I can select one https://github.com/UofT-Frosh-Orientation/orientation-website/issues/186
- [x] Button selector component should also have the text in the buttons vertically centered because when the tabs have 2 lines of text, the other height scales accordingly https://github.com/UofT-Frosh-Orientation/orientation-website/issues/187
- [ ] The Check fields breaks if it has never been filled out before (seems like a problem with local storage) https://github.com/UofT-Frosh-Orientation/orientation-website/issues/188
- [ ] Some fields should have extra checks, such as selecting a birthday in 1800s, see https://github.com/UofT-Frosh-Orientation/orientation-website/issues/178

natapokie commented on client/src/pages/Registration/RegistrationFields.jsx:
```diff
@@ -81,7 +61,7 @@ export const fields = {
       inputType: 'text',
       label: 'UtorID',
-      placeholder: 'doejohn2',
+      placeholder: 'froshbest123',
```
yep! mine has numbers 😂

MatthewGWilson commented:
Thanks for looking this over. I made most of the changes you recommended. 

- I'm thinking we should leave legal name just as a redundancy in case something comes up where frosh don't sign up for an account with the name that is in the uoft system so Kerryn recommended we ask for their legal name as part of registration as a double check but you can leave this information unused unless an issue comes up. 
- I'm leaving Jane Doe as emergency contact because it still makes sense as a place holder there I just set the placeholder for the legal name to Best Frosh so it would match the utorid of froshbest123. 
- I wasn't sure if setting -1 as the default for the radio questions was okay, I can change that if it's not but there's some questions we don't have a default that makes sense for so it would be nice to be able to leave them without default.
- For accommodation contact information I only had it open if they selected other because we already have their email and phone number, if it’s easier to organize I can change it so it opens no matter what choice they make.
- Carries 2 epi-pens isn’t a title it’s the sample input in the box so doesn’t need to be capitalized
- I think country codes not being required should be okay since for local numbers we won’t need the country code

jameskokoska commented:
> Thanks for looking this over. I made most of the changes you recommended. 
> 
> - I'm thinking we should leave legal name just as a redundancy in case something comes up where frosh don't sign up for an account with the name that is in the uoft system so Kerryn recommended we ask for their legal name as part of registration as a double check but you can leave this information unused unless an issue comes up. 
Okay so we should have 
First Name, Last Name, Preferred Name, Legal name for the Frosh model then 
> - I'm leaving Jane Doe as emergency contact because it still makes sense as a place holder there I just set the placeholder for the legal name to Best Frosh so it would match the utorid of froshbest123. 
It would probably be better to do something like John Doe and johndoe123
> - I wasn't sure if setting -1 as the default for the radio questions was okay, I can change that if it's not but there's some questions we don't have a default that makes sense for so it would be nice to be able to leave them without default.
Does it make sense for a Yes and No question to not have a response? If you want them to actively decide and have blank the default value, but not a possible value for validation, can you create a ticket so that radio buttons in the registration form with no valid index set will result in the Check (validation) failing 
> - For accommodation contact information I only had it open if they selected other because we already have their email and phone number, if it’s easier to organize I can change it so it opens no matter what choice they make.
Ok sounds good
> - Carries 2 epi-pens isn’t a title it’s the sample input in the box so doesn’t need to be capitalized
Was wondering if Epi-Pen should be capitalized, I think it's a company name or something 
Similar to the inputs title, the Epi-Pen in brackets is capitalized as the example

jameskokoska commented:
Can you also give me a list of extra error checking
For e.g. throw an error message if a birthday is set before 1900 and past today's date
This is actually pretty useful because some people forget to select the proper year for their birthday, and just leave it as the current year 

or 

Postal code having 6 characters check

There is a ticket https://github.com/UofT-Frosh-Orientation/orientation-website/issues/178
So when this gets done, we will add in what things have extra validation 


Anything that you need formatted also let me know (e.g. I made it so instagram handles auto put an @ symbol, and phone numbers are formatted)
This needs to be hand coded, the way the input component works

MatthewGWilson commented:
Here's a list that I believe is fairly comprehensive for error checking and formatting but if anyone can think of anything else it would be good to have more pairs of eyes on it.

- Birthdays should be between 1920 and 2020 (not sure if this is a perfect range to use but we don't want to exclude students who are older or younger that expected while also making sure people don't just leave it auto set on 2022)
- Utorid should be formatted to be lower case but can include numbers
- Utorid should be eight characters long
- Country code should be 1 digit long
- Phone number should be 7 digits long
- instagram should be formatted to have an @ at the beginning unless it's already there
- Emergency number / country code should have same things as phone number / country code above

I also made updates according to the feedback.

- Removed preferred name and will make sure the frosh model has first, last, and legal name
- I think best frosh might be more fun but John Doe is clearer so I switched to that
- It would be more ideal to not give the radio fields a default answer but it sounds like that'll be more work so I just assigned defaults which is fine
- You're right EpiPen is a brand name I didn't realize

jameskokoska commented:
> Here's a list that I believe is fairly comprehensive for error checking and formatting but if anyone can think of anything else it would be good to have more pairs of eyes on it.
> 
> - Birthdays should be between 1920 and 2020 (not sure if this is a perfect range to use but we don't want to exclude students who are older or younger that expected while also making sure people don't just leave it auto set on 2022)
> - Utorid should be formatted to be lower case but can include numbers
> - Utorid should be eight characters long
> - Country code should be 1 digit long
> - Phone number should be 7 digits long
> - instagram should be formatted to have an @ at the beginning unless it's already there
> - Emergency number / country code should have same things as phone number / country code above
> 

Sounds good, I added to issue https://github.com/UofT-Frosh-Orientation/orientation-website/issues/178 as a comment.

> I also made updates according to the feedback.
> 
> - Removed preferred name and will make sure the frosh model has first, last, and legal name

Sure, but preferred name is still in the Create Account page don't forget 

> - I think best frosh might be more fun but John Doe is clearer so I switched to that

Okay sounds good, but it's a bit inconsistent with the create account page
Could you submit a ticket to change that / change yourself in a PR
See
https://github.com/UofT-Frosh-Orientation/orientation-website/pull/164

> - It would be more ideal to not give the radio fields a default answer but it sounds like that'll be more work so I just assigned defaults which is fine

It's not a problem, just create a ticket so radio buttons support errors when an invalid index is supplied 

> - You're right EpiPen is a brand name I didn't realize

ashnajain48 commented:
> Here's a list that I believe is fairly comprehensive for error checking and formatting but if anyone can think of anything else it would be good to have more pairs of eyes on it.
> 
> * Birthdays should be between 1920 and 2020 (not sure if this is a perfect range to use but we don't want to exclude students who are older or younger that expected while also making sure people don't just leave it auto set on 2022)
> * Utorid should be formatted to be lower case but can include numbers
> * Utorid should be eight characters long
> * Country code should be 1 digit long
> * Phone number should be 7 digits long
> * instagram should be formatted to have an @ at the beginning unless it's already there
> * Emergency number / country code should have same things as phone number / country code above

For people outside Canada, some country codes are more than 1 digit so maybe take out the 1 digit limit

jameskokoska commented:
See https://github.com/UofT-Frosh-Orientation/orientation-website/issues/178

Oops! Don't close this PR as it isn't merged yet

jameskokoska commented:
@MatthewGWilson Please push whatever changes you have and I can do a review again, when ready
Also someone else can do a review and then it can be merged into dev

Cali0707 reviewed (approved):
This looks good to me!

MatthewGWilson commented:
I think I had screwed this up earlier by creating a fork instead of a branch. I pushed to #162 with my registration changes now so hopefully that works

jameskokoska commented:
Ok sounds good @MatthewGWilson 
You just forgot to make the PR, but I made it here: https://github.com/UofT-Frosh-Orientation/orientation-website/pull/21
